src/coreclr/vm/prestub.cpp

```diff
@@ -600,6 +600,29 @@ PCODE MethodDesc::GetPrecompiledR2RCode(PrepareCodeConfig* pConfig)
         {
             pCode = pModule->GetReadyToRunInfo()->GetEntryPoint(this, pConfig, TRUE /* fFixups */);
         }
+
+        if (!pCode)
+        {
+            AppDomain::AssemblyIterator it = SystemDomain::System()->DefaultDomain()->IterateAssembliesEx((AssemblyIterationFlags)(kIncludeLoaded|kIncludeExecution));
+            CollectibleAssemblyHolder<DomainAssembly *> pDomainAssembly;
+            while (it.Next(pDomainAssembly.This()))
```

Comment:
We do see important applications that load 1000+ assemblies. I suspect that iterating through 1000+ assemblies before JITing each method will show up as unacceptable performance regression.

Reply:
@y-yamshchikov A hashmap might reduce looping time. Please refer to FnV code.

Reply:
@HJLeee for creating hasmap we have to make global table and add new entries on every assembly loading. It will take time (may be not even big) and increase memory consumption (otherwise we'll have to change R2R format). Will recheck FNV if current approach will not give enough gain.

In case of Large Version Bubble we should to search compiled instantiation of external generic method in assembly where it's being called first. It should help with many case related to #46160. So now we're implementing this logic in `ExternalMethodFixupWorker`.